src/snowcli/utils.py

```diff
@@ -478,66 +477,59 @@ def install_packages(
-
-    # zip all files in the current directory except the ones that start with "." or are in the pack_dir
-    for file in pathlib.Path(".").glob("**/*"):
-        if (
-            not str(file).startswith(".")
-            and not file.match(f"{pack_dir}/*")
-            and not file.match(dest_zip)
-        ):
-            zipf.write(os.path.relpath(file))
-
-    if os.getenv("SNOWCLI_INCLUDE_PATHS") is not None:
-        for dir_path in os.getenv("SNOWCLI_INCLUDE_PATHS", "").split(":"):
-            directory = pathlib.Path(dir_path)
-            if directory.is_dir():
-                for file in pathlib.Path(directory).glob("**/*"):
-                    if (
-                        not str(file).startswith(".")
-                        and not file.match("*.pyc")
-                        and not file.match("*__pycache__*")
-                    ):
-                        zipf.write(file, arcname=os.path.relpath(file, directory))
-
-    # close the zip file object
-    zipf.close()
+    files_to_pack = get_list_of_files_to_pack(pack_dir, True)
+
+    with ZipFile(dest_zip, "w", ZIP_DEFLATED, allowZip64=True) as package_zip:
+        for file in files_to_pack:
+            package_zip.write(file[0], arcname=os.path.relpath(file[0], file[1]))
     return True
 
 
 def standard_zip_dir(dest_zip: str) -> bool:
-    zipf = zipfile.ZipFile(dest_zip, "w", zipfile.ZIP_DEFLATED, allowZip64=True)
-    for file in pathlib.Path(".").glob("*"):
-        if not file.match(".*"):
-            zipf.write(os.path.relpath(file))
-
-    if os.getenv("SNOWCLI_INCLUDE_PATHS") is not None:
-        for dir_path in os.getenv("SNOWCLI_INCLUDE_PATHS", "").split(":"):
-            directory = pathlib.Path(dir_path)
-            if directory.is_dir():
-                for file in pathlib.Path(directory).glob("**/*"):
-                    if (
-                        not str(file).startswith(".")
-                        and not file.match("*.pyc")
-                        and not file.match("*__pycache__*")
-                    ):
-                        zipf.write(file, arcname=os.path.relpath(file, directory))
-
-    # close the zip file object
-    zipf.close()
+    files_to_pack = get_list_of_files_to_pack(None, False)
+
+    with ZipFile(dest_zip, "w", ZIP_DEFLATED, allowZip64=True) as package_zip:
+        for file in files_to_pack:
+            package_zip.write(file[0], arcname=os.path.relpath(file[0], file[1]))
     return True
 
 
+def get_list_of_files_to_pack(
+    pack_dir: Optional[str], is_recursive: bool
+) -> List[Tuple[Path, Optional[str]]]:
+
+    files: List[Tuple[Path, Optional[str]]] = []
+
+    def filenames_filter(file: Path) -> bool:
+        return (
+            not file.name.startswith(".")
+            and not str(file).startswith(".")
+            and not file.match("*.pyc")
+            and not file.match("*__pycache__*")
+            and file not in map(lambda x: x[0], files)
```

Comment:
```suggestion
            and file not in map(lambda x: x[0], excluded_files)
```
WDYT? Also, `[f[0] for f in files]` would be probably more pythonic

Reply:
MAybe we can use `set` instead of list? It automatically resolves the duplicates. WDYT?

Reply:
I thought about it, but, when added in different steps, Files can have the same name and different relpath - therefore a simple change from list to set would potentially leave some duplicates.
And, as we`re already checking the filename, i thought adding the check there would the simplest solution.